### PR TITLE
Bump (Lib)CURL to 7.81.0

### PR DESCRIPTION
**Pull request by mkitti**
URL has changed permanently from curl.haxx.se to curl.se. Also note to depends on MbedTLS 2.28 in this script.

I would consider holding this for a bit as some potential issues between LibCURL 7.81 and Julia are investigated:
https://github.com/JuliaLang/julia/issues/43682

However, I have built Julia master with curl 7.81.0 and MbedTLS 2.28 just fine locally on Linux.

### CODE REVIEW

**Comment by mkitti** [comment]
After the investigation outlined in https://github.com/JuliaLang/Downloads.jl/issues/172#, I think we are good to merge this if there are no other objections.

**Comment by giordano** [comment]
You need first to make this work at all 🙂

File: L/LibCURL/build_tarballs.jl [hunk 1]
```diff
@@ -76,8 +76,8 @@ dependencies = [
     Dependency("nghttp2_jll"),
     # Note that while we unconditionally list MbedTLS as a dependency,
     # we default to schannel/SecureTransport on Windows/MacOS.
-    Dependency("MbedTLS_jll", v"2.24.0"),
+    Dependency("MbedTLS_jll", v"2.28.0"),
```
Comment (giordano) [inline L/LibCURL/build_tarballs.jl:79]:
```suggestion
    Dependency("MbedTLS_jll"; compat="~2.28.0"),
```

**Comment by mkitti** [comment]
I think the correct configuration option is `--without-libgsasl`. I'm not sure where `--disable-libgsasl` came from:
https://github.com/JuliaPackaging/Yggdrasil/pull/3575#issuecomment-910863020

File: L/LibCURL/build_tarballs.jl [hunk 2]
```diff
@@ -76,8 +76,8 @@ dependencies = [
     Dependency("nghttp2_jll"),
     # Note that while we unconditionally list MbedTLS as a dependency,
     # we default to schannel/SecureTransport on Windows/MacOS.
-    Dependency("MbedTLS_jll", v"2.24.0"),
+    Dependency("MbedTLS_jll"; compat="~2.28.0"),
```
Comment (jeremiahpslewis) [inline L/LibCURL/build_tarballs.jl:79]:
```suggestion
    Dependency("MbedTLS_jll"; compat="~2.28.0"),
    Dependency("LibSSH2_jll"),
```

Just reading the last remaining error, are you missing the following (above)?

Comment (mkitti) [reply L/LibCURL/build_tarballs.jl:79]: Interesting.  Why is this only needed on one platform?

Comment (giordano) [reply L/LibCURL/build_tarballs.jl:79]: No https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=15920&view=logs&j=a6122f81-e5a2-5211-451a-3f540ebd06e0&t=ff55cc42-c3ce-53e5-62b6-7d1c7f90b9e0&l=3499

**Comment by mkitti** [comment]
I believe the test is misfiring for some reason.

https://github.com/curl/curl/blob/18480f71be0685ee6becaa85c762c6b7956a2402/acinclude.m4#L1924-L1955

>   dnl this test is of course not sensible if we are cross-compiling!

So the problem is that on `LibCURL-x86_64-linux-musl`, it has failed to figure out that is cross-compiling.

**Comment by mkitti** [comment]
or perhaps this matches the description of the actual host... so we are not actually cross compiling

**Comment by giordano** [comment]
When everything else fails, [read the documentation](https://docs.binarybuilder.org/stable/troubleshooting/#General-comments)

> The best recommendation is to access the build environment and carefully read the log files generated by the build systems: it is not uncommon that build systems would only print to screen misleading error messages, and the actual problem is completely different (e.g. "library XYZ can't be found", when the problem instead is that the command they run to look for library XYZ fails for unrelated reasons, for example for a wrong compiler flag used in the check).

If you had entered the build environment and read the `config.log` file you'd have found

```
configure:33994: checking run-time libs availability
configure:34022: cc -std=gnu11 -o conftest -Werror-implicit-function-declaration -O2 -Wno-system-headers  -I/workspace/destdir/include -I/workspace/destdir/include -I/workspace/destdir/include -I/workspace/destdir/include   -L/workspace/destdir/lib -L/workspace/destdir/lib -L/workspace/destdir/lib -L/workspace/destdir/lib  conftest.c -lnghttp2  -lssh2 -lssh2 -lmbedtls -lmbedx509 -lmbedcrypto -lz  >&5
/opt/x86_64-linux-musl/bin/../lib/gcc/x86_64-linux-musl/4.8.5/../../../../x86_64-linux-musl/bin/ld: warning: libmbedtls.so.13, needed by /workspace/destdir/lib/libssh2.so, not found (try using -rpath or -rpath-link)
/opt/x86_64-linux-musl/bin/../lib/gcc/x86_64-linux-musl/4.8.5/../../../../x86_64-linux-musl/bin/ld: warning: libmbedcrypto.so.5, needed by /workspace/destdir/lib/libssh2.so, not found (try using -rpath or -rpath-link)
```

That's because our libssh2 build links to mbedtls 2.24, which is incompatible with 2.28.  Before rebuilding libcurl you _**must**_  rebuild libssh2

**Comment by jeremiahpslewis** [comment]
@giordano Thank you! what is the best way to get the `config.log` file out of the docker container (on a Mac) and read the `config.log` file in vscode?

**Comment by giordano** [comment]
Open it?

**Comment by jeremiahpslewis** [comment]
But it's in the docker container, so it doesn't get output to / isn't accessible by the local file system, right?

**Comment by mkitti** [comment]
Of course.

**Comment by giordano** [comment]
If you build with
```
julia build_tarballs.jl --debug --verbose <platform>
```
as suggested in [the very first page of the documentation](https://docs.binarybuilder.org/stable/#Manually-create-or-edit-build_tarballs.jl), when something fails you get dropped inside the build environment, where you can browse around and open all the files you want.

**Comment by jeremiahpslewis** [comment]
@giordano I'm with you, I use the build environment interface often, but find it more comfortable to scan through a file in the VSCode file viewer...maybe the `config.log` file could be written to the `build` directory to surface the file more directly? 
Screen Shot 2022-01-09 at 12 18 33

**Comment by giordano** [comment]
Have you seen the `build/` directory created in your current directory when you build?  That contains everything which is mounted in the build environment.

**Comment by jeremiahpslewis** [comment]
> Have you seen the `build/` directory created in your current directory when you build? That contains everything which is mounted in the build environment.

Thanks!! File was not showing up in the file picker so I assumed it didn't exist.

**Comment by nalimilan** [comment]
Can this be merged? It's needed by Julia for https://github.com/JuliaLang/julia/pull/43250.

**Comment by giordano** [comment]
It's failing, so no, it can't be merged

**Comment by mkitti** [comment]
Rerun ci

**Pull request merged**